packages/aws-cdk/lib/api/cxapp/cloud-assembly.ts

```diff
@@ -100,7 +107,11 @@ export class CloudAssembly {
     const patterns = sanitizePatterns(selector.patterns);
 
     if (stacks.length === 0) {
-      throw new Error('This app contains no stacks');
+      if (options.ignoreNoStacks) {
+        return new StackCollection(this, []);
+      } else {
+        throw new Error('This app contains no stacks');
+      }
```

Comment:
```suggestion
      throw new Error('This app contains no stacks');
```
No need for an `else`